include/SDL_render.h

```diff
@@ -1414,6 +1423,51 @@ extern DECLSPEC int SDLCALL SDL_RenderCopyExF(SDL_Renderer * renderer,
                                             const SDL_FPoint *center,
                                             const SDL_RendererFlip flip);
 
+/**
+ *  Render a list of triangles, optionally using a texture and indices into the vertex array
+ *  Color and alpha modulation is done per vertex (SDL_SetTextureColorMod and SDL_SetTextureAlphaMod are ignored).
+ *
+ *  \param texture      (optional) The SDL texture to use.
+ *  \param vertices     Vertices.
+ *  \param num_vertices Number of vertices.
+ *  \param indices      (optional) An array of integer indices into the 'vertices' array, if NULL all vertices will be rendered in sequential order.
+ *  \param num_indices  Number of indices.
+ *
+ *  \sa SDL_Vertex
+ *
+ *  \return 0 on success, or -1 if the operation is not supported
+ */
+extern DECLSPEC int SDLCALL SDL_RenderGeometry(SDL_Renderer *renderer,
+                                               SDL_Texture *texture,
+                                               const SDL_Vertex *vertices, int num_vertices,
+                                               const int *indices, int num_indices);
+
+/**
+ *  Render a list of triangles, optionally using a texture and indices into the vertex arrays
+ *  Color and alpha modulation is done per vertex (SDL_SetTextureColorMod and SDL_SetTextureAlphaMod are ignored).
+ *
+ *  \param texture      (optional) The SDL texture to use.
+ *  \param xy           Vertex positions
+ *  \param xy_stride    Byte size to move from one element to the next element
+ *  \param color        Vertex colors (as SDL_Color)
+ *  \param color_stride Byte size to move from one element to the next element
+ *  \param uv           Vertex normalized texture coordinates
+ *  \param uv_stride    Byte size to move from one element to the next element
+ *  \param num_vertices Number of vertices.
+ *  \param indices      (optional) An array of indices into the 'vertices' arrays, if NULL all vertices will be rendered in sequential order.
+ *  \param num_indices  Number of indices.
+ *  \param size_indice  Indice size: 1 (byte), 2 (short), 4 (int)
+ *
+ *  \return 0 on success, or -1 if the operation is not supported
+ */
+extern DECLSPEC int SDLCALL SDL_RenderGeometryRaw(SDL_Renderer *renderer,
```

Comment:
Is there much point to having this function considering all the vertex attributes are mandatory (you can't pass in nulls)? Flexibility is nice I guess, but IMO API flexibility should come from real-world use cases after determining simpler APIs can't satisfy specific needs rather than providing a kitchen sink in the hope that people will actually need everything there.

Personally I'd prefer it if just `SDL_RenderGeometry` existed - there's pros and cons to accepting an array of structures versus multiple independent arrays, but picking one of the two and sticking with it feels more in line with SDL_Render's simple design philosophy compared to lower level graphics APIs.

Reply:
Having just a single simpler / less catch-all external API will also help keep backend implementations simple too. For example the ability to memcpy an array of `SDL_Vertex` structs directly, and pass that all the way through to the GPU without looping over each vertex, is possible on several backends if the higher level API isn't too loose with its constraints.

Reply:
(uv can be null, indices can be null)
After trying to integrate the SDL_vertex with real-world libs, it appears I need to allocate/loop/re-assign the datas.
So this the common denominator when integrating with the tested libs. The simpliest if you don't want to re-alloc and re-order datas.

SDL_RenderGeometry and SDL_RenderGeometryRaw share the same backend interface. (SDL_RenderGeometry is just 5 lines)

Reply:
Can we rename this to `SDL_RenderGeometryEx` just to have a consistent naming? (we have `SDL_RenderCopyEx`)